app/models/manageiq/providers/redhat/inventory/persister/target_collection.rb

```diff
@@ -117,8 +117,6 @@ def add_vms_inventory_collections(manager_refs)
     )
     add_inventory_collection(
       infra.vm_and_template_ems_custom_fields(
-        :arel     => CustomAttribute.joins("INNER JOIN vms ON vms.id = custom_attributes.resource_id")
```

Comment:
hm, but without this, the targeted refresh will now delete all ```vm_and_template_ems_custom_fields```

so we should rather fix the query?

Reply:
Oh it isn't automatically scoped to the target?

Reply:
no, rhev is not using the new targeted refresh, so it's all defined manually here